### DISCUSSION TITLE
Blank image when music uri doesnt contain an image

### DISCUSSION THREAD

**Discussion by lukatsivilashvili** [coil-kt/coil#1384] [Q&A]
I dont know what im doing wrong, so maybe you could help me. im building a simple music player in compose that reads music from mediastore and sets artwork from uri. Everything works fine except for placeholder in a specific scenario.
AsyncImage(
            model = ImageRequest.Builder(LocalContext.current)
                .data(uri)
                .placeholder(R.drawable.ic_default_artwork)
                .crossfade(true)
                .build(),
            placeholder = painterResource(R.drawable.ic_default_artwork),
            contentDescription = "Music Art",
            contentScale = ContentScale.Fit,
        )
So heres the problem: If the mediaItem has the artwork uri but doesnt contain an image e.g music doesnt have an album art, theres nothing displayed. its as if a transparent png was placed there.

Any ideas? it is my first time with coil let alone on compose, any help would be appreciated. as a side note, i have another similar app written with xmls and glide, and the placeholder works properly.

**Comment by Jawnnypoo** [comment]
Placeholder refers to what will show when the image is in the process of loading. If the image is not available, you would want to use `error` or `fallback` instead (error for when it fails to load, fallback for when the data is null)

**Reply by lukatsivilashvili** [reply]
I'll try this out and get  back to you 👍

**Reply by lukatsivilashvili** [reply]
I also forgot to mention that there is no error nor the uri is empty. I already checked it and the uri exists. The problem is that the uri itself is "empty" not null, so i guess there is no support for that?

**Reply by Jawnnypoo** [reply]
you can just do a check and if `uri.isEmpty()`, then make it pass `null` instead of the empty string

**Reply by lukatsivilashvili** [reply]
Sorry if i'm explaining it wrong. On every song, the arturi comes back as follows: content://media/external/audio/albumart/3082221280151247212
this always holds this kind of value with different numbers. However, when the system goes to this location to load the artwork, the artwork isnt there, so nothing is set. fallback and error arent being utilised because the uri isnt null and it was successfully retrieved.
In the same scenario, as i explained i used glide's placeholder parameter and it seemed to do the work if it gives any help
